kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/kommunicate/KmAttachmentsController.java

```diff
@@ -0,0 +1,198 @@
+package com.applozic.mobicomkit.uiwidgets.kommunicate;
+
+import android.content.Context;
+import android.database.Cursor;
+import android.net.Uri;
+import android.provider.OpenableColumns;
+import android.text.TextUtils;
+
+import com.applozic.mobicomkit.api.account.user.MobiComUserPreference;
+import com.applozic.mobicomkit.api.attachment.FileClientService;
+import com.applozic.mobicomkit.api.conversation.Message;
+import com.applozic.mobicomkit.uiwidgets.AlCustomizationSettings;
+import com.applozic.mobicomkit.uiwidgets.ApplozicSetting;
+import com.applozic.mobicomkit.uiwidgets.R;
+import com.applozic.mobicomkit.uiwidgets.async.FileTaskAsync;
+import com.applozic.mobicomkit.uiwidgets.kommunicate.callbacks.PrePostUIMethods;
+import com.applozic.mobicommons.commons.core.utils.Utils;
+import com.applozic.mobicommons.file.FileUtils;
+
+import java.io.File;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * contains all methods that can be used to send, write and manage attachment [messages]
+ *
+ * @author shubham
+ * 20th September, 2019
+ */
+public class KmAttachmentsController {
+    Context context;
+    public static final String TAG = "KmAttController";
+
+    public KmAttachmentsController(Context context) {
+        this.context = context;
+    }
+
+    /**
+     * create a message object and set its attributes for a file (uri given) and return it
+     *
+     * @param uri the uri of the file
+     * @param multiSelectGalleryUpload true if this is just a image/video (one step flow)
+     * @param groupID the groupId to send to
+     * @param userID the userId to send to
+     * @param messageText the message text (only for the non-multiSelectGalleryUpload flow)
+     * @throws Exception if uri path is empty and others
+     * @return the message (messageToSend)
+     */
+    public Message putAttachmentInfo(Uri uri, boolean multiSelectGalleryUpload, int groupID, String userID, String messageText) throws Exception{
+        MobiComUserPreference userPreference = MobiComUserPreference.getInstance(context);
+            String filePath = uri.getPath();
+            if (TextUtils.isEmpty(filePath)) {
+                Utils.printLog(context, TAG, context.getResources().getString(R.string.info_file_attachment_error));
+                throw new Exception(""+R.string.info_file_attachment_error);
+            }
+            Message messageToSend = new Message();
+            if (groupID != 0) {
```

Comment:
the group id is passed as an intent extra from another activity. If no groupId is passed, the value is set to 0 by default. thats why we have thid check.
Also its more efficient to pass int etc rather than objects.

Reply:
this is wrong and confusing as what is 0, 0 is a valid number and what if 0 or any other number is a proper group id. for all message which are 1-1, server also stores and returns null everywhere and 0.

Reply:
the group id is passed as an intent extra from another activity. If no groupId is passed, the value is set to 0 by default. thats why we have thid check.
Also its more efficient to pass int etc rather than objects.

Reply:
using Integer object now